native/src/components/MapView.tsx

```diff
@@ -130,6 +130,20 @@ const MapView = ({
   const locationPermissionGrantedIcon = followUserLocation ? LocationFixedIcon : LocationNotFixedIcon
   const locationPermissionIcon = locationPermissionGranted ? locationPermissionGrantedIcon : LocationOffIcon
 
+  const zoomOnClusterPress = async (pressedLocation: Feature<Geometry, GeoJsonProperties>) => {
+    if (!mapRef.current || !pressedLocation.properties) {
+      return
+    }
+    const clusterCollection = await mapRef.current.queryRenderedFeaturesAtPoint(
+      [pressedLocation.properties.screenPointX, pressedLocation.properties.screenPointY],
+      undefined,
+      [clusterLayerId],
+    )
+    if (clusterCollection && 0 in clusterCollection.features) {
+      const feature = clusterCollection.features[0] as MapFeature
+      moveTo(feature.geometry.coordinates, bottomSheetHeight, (await mapRef.current.getZoom()) + 2)
```

Comment:
Not sure if I understand this, we are always just zooming in `2` steps more than the current zoom level? We don't have a specific zoom level we always want as default for clicking on a cluster?

Also, why do we need to query features just to zoom in to the pressed location we already moved to? And why are we calling moveTo again instead of just setting the zoom level (https://github.com/maplibre/maplibre-react-native/blob/main/docs/Camera.md#zoomtozoomlevel-animationduration)?

Reply:
well because you don't want to specify a zoom level but just zoom a little bit closer on click, since you start on a very low zoom level its a very ugly user experience to zoom directly to detail click. And you want to center the point you are clicking thats why i use `flyTo`

Reply:
Oh sorry, I confused multipoints with clusters... thanks for the explanation :)